src/parser/SparqlParser.cpp

```diff
@@ -512,15 +520,27 @@ void SparqlParser::addFilter(const string& str, vector<SparqlFilter>* _filters,
           // an expensive regex filter.
           bool isSimple = true;
           bool escaped = false;
+
+          std::vector<bool> regexControlChars(sizeof(char), false);
```

Comment:
`sizeof(char)` is 1, I think you're looking for `std::numeric_limists<unsigned char>::max()`. Also this still allocates. If you use `std::array` instead of `std::vector` the allocation becomes a simple addittion on the stack pointer but it still leaves us with setting the bad char elements on each call. Instead we could make this a member variable of `SparqlParser` and only set it during construction.

Alternatively I think the linear search isn't really all that bad with searching in a very small string so the old variant is probably just as fast. Though I would use a `std::array<char> = {'[', …};` then since I'm not sure if all chars fit into the small string op and again, `std::array` would allocate on the stack

Reply:
Yeah, that's a nice solution, I might have turned this into a bit of a useless detour for such a small thing but thanks for your effort!